src/components/select_team.js

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component, PropTypes} from 'react';
+
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {Image, StyleSheet, Picker, View} from 'react-native';
+import {Actions as Routes} from 'react-native-router-flux';
+import _ from 'lodash';
+
+import * as teamActions from 'actions/teams';
+import ErrorText from 'components/error_text';
+import logo from 'images/logo.png';
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        flexDirection: 'column',
+        alignItems: 'center',
+        paddingTop: 200,
+        backgroundColor: 'white'
+    },
+    logo: {
+        marginBottom: 10
+    }
+});
+
+const propTypes = {
+    teams: PropTypes.object.isRequired,
+    actions: PropTypes.object.isRequired
+};
+
+class SelectTeam extends Component {
+    static propTypes = propTypes;
+
+    componentWillMount() {
+        this.props.actions.fetchTeams();
+    }
+
+    componentWillReceiveProps(props) {
+        if (props.teams.current_team_id && !this.props.teams.current_team_id) {
+            Routes.ChannelsList(); // eslint-disable-line new-cap
+        }
+    }
+
+    render() {
+        return (
+            <View style={styles.container}>
+                <Image
+                    style={styles.logo}
+                    source={logo}
+                />
+                <Picker
+                    style={{width: 300}}
+                    onValueChange={(team) => {
+                        this.props.actions.selectTeam(team);
+                    }}
+                >
+                    {_.map(this.props.teams.data, (team) => (
```

Comment:
This doesn't need to be changed right now, but this seems like it could just be `this.props.teams.data.map...`. Since we've got babel, I'd rather not include lodash just to polyfill stuff that is already supported, but I'm fine if we need it for other stuff.